monitoring/monitoring_on.go

```diff
@@ -0,0 +1,67 @@
+// +build monitoring
+
+package monitoring
+
+import (
+	"fmt"
+	"net/http"
+
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Enabled signifies whether the monitoring tag was specified when building lnd
+// and whether to automatically export Prometheus metrics.
+const Enabled = true
+
+// GetServerOpts returns the set of interceptors for macaroon authentication
+// (if enabled) and Prometheus monitoring.
+func GetServerOpts(macService *macaroons.Service, perms map[string][]bakery.Op,
+	serverOpts []grpc.ServerOption) []grpc.ServerOption {
+
+	var (
+		unaryInterceptor  grpc.UnaryServerInterceptor
+		streamInterceptor grpc.StreamServerInterceptor
+	)
+
+	// If macaroons aren't disabled (a non-nil service), then we'll set up
+	// our set of interceptors which will allow us handle the macaroon
+	// authentication in a single location. If monitoring is enabled, we'll
+	// also set up interceptors for the Prometheus exporter. If both are
+	// enabled, we'll chain the interceptors together.
+	switch macService {
+	case nil:
+		unaryInterceptor = grpc_prometheus.UnaryServerInterceptor
+		streamInterceptor = grpc_prometheus.StreamServerInterceptor
+	default:
+		unaryInterceptor = grpc_middleware.ChainUnaryServer(
+			macService.UnaryServerInterceptor(perms),
+			grpc_prometheus.UnaryServerInterceptor,
+		)
+		streamInterceptor = grpc_middleware.ChainStreamServer(
+			macService.StreamServerInterceptor(perms),
+			grpc_prometheus.StreamServerInterceptor,
+		)
+
+	}
+
+	return append(serverOpts,
+		grpc.UnaryInterceptor(unaryInterceptor),
+		grpc.StreamInterceptor(streamInterceptor),
+	)
+}
+
+// ExportPrometheusMetrics sets server options, registers gRPC metrics and
+// launches the Prometheus exporter on the specified address.
+func ExportPrometheusMetrics(grpcServer *grpc.Server, listenAddr string) {
+	if listenAddr == "" {
+		listenAddr = "localhost:8989"
```

Comment:
would think that this should be set where the other defaults are handled?

if we define an `lncfg.DefaultPrometheus()` for both on and off, one can be empty and the other can contain this value